contrib/node/src/python/pants/contrib/node/subsystems/node_distribution.py

```diff
@@ -35,25 +39,40 @@ def register_options(cls, register):
       register('--version', advanced=True, default='6.9.1',
                help='Node distribution version.  Used as part of the path to lookup the '
                     'distribution with --binary-util-baseurls and --pants-bootstrapdir')
+      register('--package-manager', advanced=True, default='npm',
+               help='Default package manager config for repo. Should be one of [npm,yarnpkg]')
+      register('--yarnpkg-version', advanced=True, default='v0.19.1',
+               help='Yarnpkg version. Used for binary utils')
 
     def create(self):
       # NB: create is an instance method to allow the user to choose global or scoped.
       # It's not unreasonable to imagine multiple Node versions in play; for example: when
       # transitioning from the 0.10.x series to the 0.12.x series.
       binary_util = BinaryUtil.Factory.create()
       options = self.get_options()
-      return NodeDistribution(binary_util, options.supportdir, options.version)
+      # for o in options:
+      #   logger.debug('NodeDistrybution.Factory.create options[%s]: %s', o, options[o])
+      return NodeDistribution(
+        binary_util, options.supportdir, options.version,
+        package_manager=options.package_manager, yarnpkg_version=options.yarnpkg_version)
 
   @classmethod
   def _normalize_version(cls, version):
     # The versions reported by node and embedded in distribution package names are 'vX.Y.Z' and not
     # 'X.Y.Z'.
     return version if version.startswith('v') else 'v' + version
 
-  def __init__(self, binary_util, relpath, version):
+  def __init__(self, binary_util, relpath, version, package_manager, yarnpkg_version):
     self._binary_util = binary_util
     self._relpath = relpath
     self._version = self._normalize_version(version)
+    package_manager = 'yarnpkg' if package_manager == 'yarn' else package_manager
+    if package_manager not in ['npm', 'yarnpkg']:
+      raise RuntimeError('Unknown package manager: %s' % package_manager)
```

Comment:
If you make it a class-level list, then (re-)checking it using the list is fine.